### PR TITLE
Feature/215 cleanup trains

### LINKED ISSUE #215: Cleanup trains
## User stories
As a developer I want a better train-API in order to have a consistent api

## Dependencies
Depends on: 
 - #26 
 - #28 

## Acceptance Criteria
- [x] `SimulationObjectUpdatingComponent` needs a property on the trains
  - [x] tests
- [x] The `track` property should return a `Track`, not a string, and should get updated regularly (at least on every access)
  - [x] tests

### PR DESCRIPTION
**Pull request by SaturnHafen**
Fixes #215 

Updates the train to return domain objects

## PR checklist

- [x] Acceptance criteria fulfilled
- [x] Additional features are tested
- [x] Docs (code, wiki & diagrams) updated
- [ ] Breaking changes anounced
- [ ] Dev-branch has been merged into local branch to resolve conflicts
- [ ] Tests and linter have passed AFTER local merge
- [ ] Another dev reviewed and approved

### CODE REVIEW

File: src/wrapper/simulation_objects.py [hunk 1]
```diff
@@ -298,6 +299,12 @@ def track(self) -> Track:
 
         :return: The current track the train is on
         """
+        if self._track is None or self._track.identifier != self._track_id:
```
Comment (CR1337) [inline src/wrapper/simulation_objects.py:302]: Is it possible that there is no `item` in `self.updater.tracks` with `item.identifier == self._track_id`?
If yes, what happens then?

Comment (SaturnHafen) [reply src/wrapper/simulation_objects.py:302]: Then this method will crash (which will happen right now if called incorrectly). But when the sim-objects are loaded on the first tick, this should not happen anymore.

Comment (CR1337) [reply src/wrapper/simulation_objects.py:302]: OK, then go for it :)

**Review by CR1337 — APPROVED** [review]
:+1:

**Pull request merged**